aea/helpers/pipe.py

```diff
@@ -0,0 +1,297 @@
+
+    def __init__(self, loop: Optional[AbstractEventLoop] = None):
+        self.logger = _default_logger
+        self._timeout = 0
+        self._server = None  # type: Optional[asyncio.AbstractServer]
+        self._connected = None  # type: Optional[asyncio.Event]
+        self._reader = None  # type: Optional[asyncio.StreamReader]
+        self._writer = None  # type: Optional[asyncio.StreamWriter]
+
+        s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        s.bind(("", 0))
+        s.listen(1)
+        self._port = s.getsockname()[1]
+        s.close()
+
+    async def connect(self, timeout=PIPE_CONN_TIMEOUT) -> bool:
+        self._connected = asyncio.Event()
+        self._timeout = timeout if timeout > 0 else 0
+        self._server = await asyncio.start_server(
+            self._handle_connection, host="127.0.0.1", port=self._port
+        )
+        assert self._server.sockets is not None
+        self._port = self._server.sockets[0].getsockname()[1]
+        self.logger.debug("socket pipe setup {}".format(self._port))
+
+        try:
+            await asyncio.wait_for(self._connected.wait(), self._timeout)
+        except asyncio.TimeoutError:
+            return False
+
+        self._server.close()
+        await self._server.wait_closed()
+
+        return True
+
+    async def _handle_connection(
+        self, reader: asyncio.StreamReader, writer: asyncio.StreamWriter
+    ):
+        assert self._connected is not None
+        self._connected.set()
+        self._reader = reader
+        self._writer = writer
+
+    async def write(self, data: bytes) -> None:
+        assert self._writer is not None
+        size = struct.pack("!I", len(data))
+        self._writer.write(size)
+        self._writer.write(data)
+        await self._writer.drain()
+
+    async def read(self) -> Optional[bytes]:
+        self.logger.debug("Reading pipes...")
+        assert self._reader is not None
+        try:
+            self.logger.debug("Waiting for messages...")
+            buf = await self._reader.readexactly(4)
+            if not buf:  # pragma: no cover
+                return None
+            size = struct.unpack("!I", buf)[0]
+            data = await self._reader.readexactly(size)
+            if not data:  # pragma: no cover
+                return None
+            return data
+        except asyncio.streams.IncompleteReadError as e:
+            self.logger.info(
+                "Connection disconnected while reading from node ({}/{})".format(
+                    len(e.partial), e.expected
+                )
+            )
+            return None
+
+    @property
+    def in_path(self) -> str:
+        return str(self._port)
+
+    @property
+    def out_path(self) -> str:
+        return str(self._port)
+
+
+class PosixNamedPipe(LocalPortablePipe):
+    """
+    Interprocess communication implementation using Posix named pipes
+    """
+
+    def __init__(self, loop: Optional[AbstractEventLoop] = None):
```

Comment:
if you add an optional logger keyword argument in constructor we can pass the logger from the connection